Tasks/AzureFunctionAppV1/deploymentProvider/ConsumptionWebAppDeploymentProvider.ts

```diff
@@ -30,6 +38,83 @@ export class ConsumptionWebAppDeploymentProvider extends AzureRmWebAppDeployment
         }
         return storageData;
     }
+
+    private async uploadPackage(storageDetails, deployPackage) : Promise<string> {
+        let defer = Q.defer<string>();
+        let storageAccount = storageDetails["AccountName"];
+        let storageKey = storageDetails["AccountKey"];
+        const blobService = azureStorage.createBlobService(storageAccount, storageKey);
+
+        const containerName: string = 'azure-pipelines-deploy';
+        const blobName: string = `package_${Date.now()}.zip`;
+        let fileName;
+
+        switch(deployPackage.getPackageType()){
+            case PackageType.folder:
+                let tempPackagePath = webCommonUtility.generateTemporaryFolderOrZipPath(tl.getVariable('AGENT.TEMPDIRECTORY'), false);
+                let archivedWebPackage;
+                try {
+                    archivedWebPackage = await zipUtility.archiveFolder(deployPackage.getPath(), "", tempPackagePath);
+                }
+                catch(error) {
+                    defer.reject(error);
+                }
+                tl.debug("Compressed folder into zip " +  archivedWebPackage);
+                fileName = archivedWebPackage;
+            break;
+            case PackageType.zip:
+                fileName = deployPackage.getPath();
+            break;
+            default:
+                throw new Error(tl.loc('Invalidwebapppackageorfolderpathprovided', deployPackage.getPath()));
+        }
+
+        blobService.createContainerIfNotExists(containerName, error => {
+            if (error){
+                defer.reject(error);
+            }
+
+            //upoading package
+            blobService.createBlockBlobFromLocalFile(containerName, blobName, fileName, (error, result) => {
+                if (error) {
+                    defer.reject(error);
+                }
+
+                //generating SAS URL
+                let startDate = new Date();
+                let expiryDate = new Date(startDate);
+                expiryDate.setFullYear(startDate.getUTCFullYear() + 1);
```

Comment:
Can we find what Az cli is doing in this case?

Reply:
should user have choice to configure the expiry date?

Reply:
Azure Functions core tools has an expitry time of [10 years](https://github.com/Azure/azure-functions-core-tools/blob/dab81f32ae4405c8f747efa23eb18273b6f98a51/src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs#L450). Will check with Web App team on this.

Regarding the second question, I had the same thought of exposing it as secondary variable (release variable). But, planned to accomodate once customers ask for it.